did-method-prism/shared/src/main/scala/fmgp/prism/CardanoTransactionMetadataPrismContent.scala

```diff
@@ -0,0 +1,21 @@
+package fmgp.prism
+
+import zio.json._
+import proto.prism.PrismObject
+
+case class CardanoTransactionMetadataPrismContent(c: Seq[String], v: Int)
+object CardanoTransactionMetadataPrismContent {
+  given decoder: JsonDecoder[CardanoTransactionMetadataPrismContent] =
+    DeriveJsonDecoder.gen[CardanoTransactionMetadataPrismContent]
+  given encoder: JsonEncoder[CardanoTransactionMetadataPrismContent] =
+    DeriveJsonEncoder.gen[CardanoTransactionMetadataPrismContent]
+  def fromJson(content: String) = {
+    content.fromJson[CardanoTransactionMetadataPrismContent] match
+      case Left(error)  => Left(s"Fail to parse CardanoTransactionMetadataPrismContent: $error")
+      case Right(value) =>
+        // Unclear encoding in the transaction metadata '0x' https://github.com/input-output-hk/prism-did-method-spec/issues/66
+        val hex = if (value.c.forall(_.startsWith("0x"))) value.c.map(_.drop(2)).mkString else value.c.mkString
```

Comment:
FIXME!

We need to fetch metadata as CBOR and not as JSON since we lose information on the Data types!

We should only consider a valid byte string of the type `byte string 0x58` instead of `text 0x78` https://datatracker.ietf.org/doc/html/rfc8949

https://github.com/input-output-hk/prism-did-method-spec/issues/66

Reply:
Done